### PR TITLE
Adding data connector ASimDnsActivityLogs

### PR DESCRIPTION
**Pull request by v-sabiraj**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: Solutions/Windows Server DNS/Data Connectors/template_ASimDnsActivityLogs.JSON [hunk 1]
```diff
@@ -0,0 +1,80 @@
+{
+    "id": "ASimDnsActivityLogs",
```
Comment (oshezaf) [inline Solutions/Windows Server DNS/Data Connectors/template_ASimDnsActivityLogs.JSON:2]: The id is misleading as the table is not intended to be used just by this connector. He can create problems in the future.

File: Solutions/Windows Server DNS/Data Connectors/template_ASimDnsActivityLogs.JSON [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    "title": "Windows DNS Events via AMA",
+    "publisher": "Microsoft",
+    "descriptionMarkdown": "The Windows DNS log connector allows you to easily filter and stream all analytics logs from your Windows DNS servers to your Microsoft Sentinel workspace using the Azure Monitoring agent (AMA). Having this data in Microsoft Sentinel helps you identify issues and security threats such as:\n- Trying to resolve malicious domain names.\n- Stale resource records.\n- Frequently queried domain names and talkative DNS clients.\n- Attacks performed on DNS server.\n\nYou can get the following insights into your Windows DNS servers from Microsoft Sentinel:\n- All logs centralized in a single place.\n- Request load on DNS servers.\n- Dynamic DNS registration failures.\n\nWindows DNS events are supported by Advanced SIEM Information Model (ASIM) and stream data into the ASimDnsActivityLogs table. [Learn more](https://docs.microsoft.com/azure/sentinel/normalization)",
+    "graphQueries": [
+        {
+            "metricName": "Total data received",
+            "legend": "ASimDnsActivityLogs",
+            "baseQuery": "ASimDnsActivityLogs\n| where EventProduct == \"DNS Server\""
```
Comment (oshezaf) [inline Solutions/Windows Server DNS/Data Connectors/template_ASimDnsActivityLogs.JSON:10]: The EventProduct is generic and might be used by other sources using this table. Please add also a condition on EventVendor to the where clauses used.

File: Solutions/Windows Server DNS/Data Connectors/template_ASimDnsActivityLogs.JSON [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            "query": "ASimDnsActivityLogs\n| where EventProduct == \"DNS Server\"\n            | where EventType == \"Query\" and EventResult == \"Failure\"\n| summarize count() by SrcIpAddr, EventResultDetails"
+        },
+        {
+            "description": "Count the number of queries that return NXDOMAIN, indicating that the queries domain name does not exist, and compares the count to a threshold of 100.",
+            "query": "ASimDnsActivityLogs\n| where EventProduct == \"DNS Server\"\n            | where EventResultDetails == \"NXDOMAIN\"\n| summarize c=count() by SrcIpAddr\n| where c > 100"
+        }
+    ],
+    "connectivityCriterias": [
+        {
+            "type": "ASimDnsActivityLogs",
```
Comment (oshezaf) [inline Solutions/Windows Server DNS/Data Connectors/template_ASimDnsActivityLogs.JSON:29]: I am not sure how connectivityCriterias works, but if this is used to check connectivity, we need to add the condition for EventProduct and EventVendor also here.

**Review by oshezaf — CHANGES_REQUESTED** [review]

**Comment by v-sabiraj** [comment]
Hey @oshezaf, the data connector is already there in the portal, we have just downloaded that and added in solution. Thanks.

**Review by v-vdixit — DISMISSED** [review]

File: Solutions/Windows Server DNS/Data Connectors/template_ASimDnsActivityLogs.JSON (hunk 2 above)
Comment (v-sabiraj) [reply Solutions/Windows Server DNS/Data Connectors/template_ASimDnsActivityLogs.JSON]: Hey @oshezaf, the query is updated as per this pr (https://dev.azure.com/msazure/One/_git/ASI-Portal/pullrequest/7562852?_a=files&path=/src/ASIExtension/Client/Data/DataConnectorsV3/Connectors/AmaConnectors/ASimDnsActivityLogsDataModel.ts), thanks.

File: Solutions/Windows Server DNS/Data Connectors/template_ASimDnsActivityLogs.JSON (hunk 3 above)
Comment (v-sabiraj) [reply Solutions/Windows Server DNS/Data Connectors/template_ASimDnsActivityLogs.JSON:29]: Hi @oshezaf, as per the discussion the connectivityCriterias will be updated at once for all connectors.

image

**Comment by v-sabiraj** [comment]
@oshezaf, can you please approve the changes, thanks.

**Comment by oshezaf** [comment]
While I understand that the connector ID cannot be changed even if incorrect, can you remind me why aren't we changing the incorrect connectivity query?

**Review by oshezaf — DISMISSED** [review]

**Review by v-dvedak — APPROVED** [review]

**Review by v-vdixit — APPROVED** [review]
Looks good!

**Pull request merged**